google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITBatchDmlTest.java

```diff
@@ -189,4 +193,47 @@ public void errorBatchDmlAlreadyExist() {
       }
     }
   }
+
+  @Test
+  public void largeBatchDml() {
+    List<Statement> stmts = new LinkedList<>();
+    for (int i = 0; i < 40; i++) {
+      stmts.add(Statement.of("INSERT INTO T (k, v) VALUES ('boo" + i + "', " + i + ");"));
+    }
+
+    for (int i = 0; i < 30; i++) {
+      stmts.add(Statement.of("DELETE FROM T WHERE T.K = 'boo" + i + "';"));
+    }
+
+    final TransactionCallable<long[]> callable = transaction -> transaction.batchUpdate(stmts);
+    TransactionRunner runner = client.readWriteTransaction();
+    long[] rowCounts = runner.run(callable);
+    assertThat(rowCounts.length).isEqualTo(10);
```

Comment:
It surprises me if this actually succeeds. The length should be 70. The fact that the last 30 statements delete 30 of the 40 rows that were inserted should still be counted as a statement that is executed, and the update count of each statement should be 1.

nit: prefer `assertEquals` over `assertThat`